Makefile/Makefile.sublime-syntax

```diff
@@ -604,33 +605,53 @@ contexts:
   variable-sub-common:
     - match: ':'
       scope: punctuation.definition.substitution.makefile
+      set: variable-substitution-reference
+    - include: highlight-percentage-sign
+    - include: variable-substitutions
+
+  variable-substitution-reference:
+    - match: '[)}]'
+      scope: keyword.other.block.end.makefile
+      pop: true
     - match: =
-      scope: punctuation.definition.assignment.makefile
+      scope: keyword.operator.assignment.makefile
     - include: highlight-percentage-sign
     - include: variable-substitutions
 
   variable-substitutions:
     - include: function-invocations
-    - match: \$\$?\(
+    - match: \$\(
       scope: keyword.other.block.begin.makefile
       push:
-        - meta_scope: variable.parameter.makefile
+        - meta_scope: meta.interpolation.makefile
```

Comment:
If we introduce `meta.interpolation`, which is very much appriciated, should we also clear `string` scope in embedded shell scripts?

I created a prototype a while ago, to include `variable-interpolations` into _Makefile Shell_ `string-prototype` context to do so:

```
  variable-interpolations:
    - match: (?=\$\$?[({@%<?^+|*[:alpha:]])
      push: variable-interpolation-body

  variable-interpolation-body:
    - clear_scopes: 1
    - meta_scope: meta.interpolation.makefile
    - include: function-invocations
    - include: variable-substitutions
    - include: immediately-pop
```

It may however require some tweaks to handle new escapes well.